docs/tutorials/data_provider.md

```diff
@@ -62,6 +62,83 @@ impl AdditiveIdentity {
 }
 ```
 
+## Loading Additional Data at Runtime
+
+A key feature of ICU4X is the ability to download data dynamically. A key use case is to add additional locales at runtime.
+
+Dynamic data loading can currently be performed in userland. A future core library API may provide this functionality; please submit feedback in [#2985](https://github.com/unicode-org/icu4x/issues/2985).
+
+```rust
+use icu_provider_adapters::either::EitherProvider;
+use icu_provider_adapters::fallback::LocaleFallbackProvider;
+use icu_provider_adapters::fork::ForkByKeyProvider;
+use icu_provider_adapters::fork::MultiForkByErrorProvider;
+use icu_provider_adapters::fork::predicates::MissingLocalePredicate;
+use icu_provider_blob::BlobDataProvider;
+use icu_provider_fs::FsDataProvider;
+use icu_provider::DataLocale;
+use icu_provider::hello_world::HelloWorldFormatter;
+use icu::locid::locale;
+use icu::locid::subtags::Language;
+use std::path::PathBuf;
+use writeable::Writeable;
+
+// Our growable data provider will be a fallback-enabled ForkByKeyProvider forking between a
+// BlobDataProvider with static core data and a MultiForkByErrorProvider with runtime-loaded data.
+type GrowableDataProvider = LocaleFallbackProvider<
+    ForkByKeyProvider<
+        BlobDataProvider,
+        MultiForkByErrorProvider<FsDataProvider, MissingLocalePredicate>
+    >
+>;
+
+// Create the empty MultiForkByErrorProvider:
+let multi_lang_provider = MultiForkByErrorProvider::new_with_predicate(
+    vec![],
+    MissingLocalePredicate
+);
+
+// Put together the rest of the data pipeline:
+static CORE_DATA: &[u8] = include_bytes!("../../../provider/adapters/tests/data/fallback.postcard");
```

Comment:
Thought: Another way to do this part, a bit more efficient, would be to create a LocaleFallbacker from a completely separate BakedDataProvider.